src/views/blockDetails/blockBody/index.tsx

```diff
@@ -0,0 +1,236 @@
+/* eslint-disable react/jsx-no-bind */
+import { useToggleVisibility } from '@pivanov/use-toggle-visibility';
+import { formatDistanceToNowStrict } from 'date-fns';
+import { useState } from 'react';
+
+import { Icon } from '@components/icon';
+import { ModalJSONViewer } from '@components/modals/modalJSONViewer';
+import { cn } from '@utils/helpers';
+
+import type {
+  IMappedBlockBody,
+  IMappedBlockEvent,
+  IMappedBlockExtrinsic,
+} from '@custom-types/block';
+
+interface BlockBodyProps {
+  bodyData: IMappedBlockBody;
+  blockNumber: number;
+}
+
+interface ButtonProps {
+  children: React.ReactNode;
+  isActive: boolean;
+  type: string;
+  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+const Button = (props: ButtonProps) => {
```

Comment:
unnecessary extraction of component we have only 2 buttons ... and children is simple text ...